CMIP7/esm1p6/atmosphere/ghg/cmip7_HI_ghg_generate.py

```diff
@@ -58,9 +58,50 @@ def load_cmip7_hi_ghg_mmr(args, ghg):
     return ghg_mmr_list
 
 
-def cmip7_hi_ghg_patch(ghg_mmr_dict):
+def read_namelists_lines_up_to(namelists_filepath, exclude_str):
     """
-    Patch the greenhouse gas variables in the clmchfcg namelist
+    Read lines from namelists_filepath up to but not including
+    the line containing exclude_str.
+    """
+    if not namelists_filepath.exists():
+        raise FileNotFoundError(
+            f"Namelist file {namelists_filepath} does not exist"
+        )
+    # Read the atmosphere/namelists file up to but not including
+    # the clmchfcg namelist.
+    namelists_str = ""
+    with open(namelists_filepath) as namelists_file:
+        namelists_line = namelists_file.readline()
+        while namelists_line and (
+            "&" + exclude_str not in namelists_line.lower()
+        ):
+            namelists_str += namelists_line
+            namelists_line = namelists_file.readline()
+    return namelists_str
+
+
+def format_namelist(
+    namelist,
+    float_format="13.6e",
+    end_comma=True,
+    false_repr=".FALSE.",
+    true_repr=".TRUE.",
+    uppercase=True,
+):
+    """
+    Change the namelist formatting to the preferred format.
+    """
+    namelist.float_format = float_format
+    namelist.end_comma = end_comma
+    namelist.false_repr = false_repr
+    namelist.true_repr = true_repr
+    namelist.uppercase = uppercase
```

Comment:
Are these values specific to this use-case? Do we think we will reuse this function? 
If this is a specific case that is likely not going to be needed elsewhere, I would simplify by removing the function arguments, leaving only `namelist`:

```suggestion
def format_namelist(namelist):
    """
    Change the namelist formatting to the preferred format.
    """
    namelist.float_format = "13.6e"
    namelist.end_comma = True
    namelist.false_repr = ".FALSE."
    namelist.true_repr = ".TRUE."
    namelist.uppercase = True
```